labs-ios-starter/CityNetworkClient.swift

```diff
@@ -91,22 +89,119 @@ class CityNetworkClient {
 //    {
 //      "msg": "crime score",
 //      "score": 4,
 //    }
+    
+    func fetchCrimeScore(forCity city: String, completion: @escaping (Result<CrimeScore, NetworkError>) -> Void) {
+        
+        let crimeScoreCityURL = baseDSURL.appendingPathComponent("crime_scr/\(city)")
+        
+        URLSession.shared.dataTask(with: crimeScoreCityURL) { (data, _, error) in
+            if let error = error {
+                print("Error \(error)")
+            }
+            
+            guard let data = data else {
+                print("No Data Returned from API")
+                return
+            }
+            
+            do {
+                let crimeScore = try self.jsonDecoder.decode(CrimeScore.self, from: data)
+                completion(.success(crimeScore))
+            } catch {
+                print("Error unable to decode the data")
+                completion(.failure(.noDecode))
+            }
+        }.resume()
+    }
 //
 //    // airquality score
 //    {
 //      "msg": "air quality score",
 //      "score": 5,
 //    }
+    
+    func fetchAirQuality(forCity city: String, completion: @escaping (Result<AirQuality, NetworkError>) -> Void) {
+        
+        let airQualityCityURL = baseDSURL.appendingPathComponent("air_qual_scr/\(city)")
+        
+        URLSession.shared.dataTask(with: airQualityCityURL) { (data, _, error) in
+            if let error = error {
+                print("Error \(error)")
+            }
+            
+            guard let data = data else {
+                print("No Data Returned from API")
+                return
+            }
+            
+            do {
+                let airQuality = try self.jsonDecoder.decode(AirQuality.self, from: data)
+                completion(.success(airQuality))
+            } catch {
+                print("Error unable to decode the data")
+                completion(.failure(.noDecode))
+            }
+        }.resume()
+    }
 //
 //    // overall score
 //    {
 //      "msg": "overall quality of life score",
 //      "score": 4,
 //    }
     
+    func fetchLifeScore(forCity city: String, completion: @escaping (Result<LifeScore, NetworkError>) -> Void) {
+        
+        let averageScoreCityURL = baseDSURL.appendingPathComponent("city_scr/\(city)")
+        
+        URLSession.shared.dataTask(with: averageScoreCityURL) { (data, _, error) in
+            if let error = error {
+                print("Error \(error)")
+            }
+            
+            guard let data = data else {
+                print("No Data Returned from API")
+                return
+            }
+            
+            do {
+                let lifeScore = try self.jsonDecoder.decode(LifeScore.self, from: data)
+                completion(.success(lifeScore))
+            } catch {
+                print("Error unable to decode the data")
+                completion(.failure(.noDecode))
+            }
+        }.resume()
+    }
+    
 }
```

Comment:
I think generics would help eliminate the duplication with all of these functions. These are fine for now, but later maybe we can make a generic fetching function that would replace these.